project-chip/connectedhomeip
# Fix incorrect return codes in door lock cluster

#### Problem
* Fixes #19622
* Fixes #19563

#### Change overview
* Now door lock cluster returns generic failure instead of NOT_FOUND when working with Week Day/Year Day credentials.
* Door lock now checks the enum ranges for Set Credential, Set User and Set Holiday Schedule commands.
* Refactor the door lock cluster a little bit

#### Testing
* Run DL_LockUnlock, DL_Schedules and DL_UsersAndCredentials tests
* Run commands from linked issues to make sure they return appropriate error codes
